lib/LaTeXML/Engine/TeX_Glue.pool.ltxml

```diff
@@ -26,19 +26,22 @@ use LaTeXML::Package;
 # \vskip            c  inserts vertical glue in a vertical list.
 # \unskip           c  removes a glue item from the current list.
 
-# a candidate for use by \hskip, \hspace, etc... ?
+our @spaces = (    # Spaces to fake spacing, with width in ems
+  [0.167, "\x{2006}"], [0.222, "\x{2005}"], [0.278, "\x{2004}"],
+  [0.333, UTF(0xA0)],  [0.5,   "\x{2002}"], [1.0,   "\x{2003}"]);
+
+# String of spacing chars with width roughly equivalent to $dimen
 sub DimensionToSpaces {
   my ($dimen) = @_;
   my $fs      = LookupValue('font')->getSize;    # 1 em
-  my $pt      = $dimen->ptValue;
-  my $ems     = $pt / $fs;
-  if    ($ems < 0.01) { return; }
-  elsif ($ems < 0.17) { return "\x{2006}"; }     # 6/em
-  elsif ($ems < 0.30) { return "\x{2005}"; }     # 4/em
-  elsif ($ems < 0.40) { return "\x{2004}"; }     # 3/em (same as nbsp?)
-  else {
-    my $n = int(($ems + 0.3) / 0.333);           # 10pts per space...?
-    return (UTF(0xA0) x $n); } }
+  my $ems     = $dimen->ptValue / $fs;
+  my $s       = '';
+  for (my $i = $#spaces ; ($i >= 0) && ($ems > 0) ; $i--) {
+    my $w = $spaces[$i][0];
+    if ($ems + 0.01 > $w) {
+      my $n = int(($ems + 0.01) / $w);
+      $ems -= $n * $w; $s .= $spaces[$i][1] x $n; } }
+  return $s; }
```

Comment:
Currently this loop would avoid adding any character if the requested dimension is narrower than `0.167em`, is that correct? Just wondering if there are legitimate cases where someone wants an even smaller amount of space. The hair space (U+200A) could fill in smaller gaps. I've seen one mention it is traditionally 1/16em or 0.0625.

Reply:
Indeed, my revision makes it easy to add more spacing chars to the mix, but I didn't actively hunt down any extras. You have a list you'd like to include?

Reply:
I do not have a list really. More of a learning-as-I-review process.